integration-tests/api/__tests__/store/order-edit.js

```diff
@@ -186,4 +186,82 @@ describe("[MEDUSA_FF_ORDER_EDITING] /store/order-edits", () => {
       expect(response.data.order_edit.confirmed_by).not.toBeDefined()
     })
   })
+
+  describe("POST /store/order-edits/:id/decline", () => {
+    let declineableOrderEdit
+    let declinedOrderEdit
+    let confirmedOrderEdit
+    beforeEach(async () => {
+      await adminSeeder(dbConnection)
+
+      declineableOrderEdit = await simpleOrderEditFactory(dbConnection, {
+        id: IdMap.getId("order-edit-1"),
+        created_by: "admin_user",
+      })
+
+      declinedOrderEdit = await simpleOrderEditFactory(dbConnection, {
+        id: IdMap.getId("order-edit-2"),
+        created_by: "admin_user",
+        declined_at: new Date(),
+      })
+
+      confirmedOrderEdit = await simpleOrderEditFactory(dbConnection, {
+        id: IdMap.getId("order-edit-3"),
+        created_by: "admin_user",
+        confirmed_at: new Date(),
+      })
+    })
+
+    afterEach(async () => {
+      const db = useDb()
+      return await db.teardown()
+    })
+
+    it("declines an order edit", async () => {
+      const api = useApi()
+      const result = await api.post(
+        `/store/order-edits/${declineableOrderEdit.id}/decline`,
+        {
+          declined_reason: "wrong color",
+        }
+      )
+
+      expect(result.status).toEqual(200)
+      expect(result.data.order_edit).toEqual(
+        expect.objectContaining({ status: "declined" })
+      )
+    })
+
+    it("fails to decline an already declined order edit", async () => {
```

Comment:
question: we stated that the declined should be idempotent [here](https://www.notion.so/medusajs/Order-Editing-6939ab89af9948688968c9387f98bab3#574871de0d5f4dfea9c6e99c0be027db) so either we make it idempotent or did a discussion happen to change that?

Reply:
nope, I just hadn't looked through the rfc thoroughly enough 😄 then only question then is, what happens to the reason that has been passed along to the decline endpoint? 

For now I have opted for your suggested solution below and just returned the already declined order edit

Reply:
would we add the idempotent feature later or should we let it down ?